src/vs/workbench/contrib/welcomeGettingStarted/browser/startupPage.ts

```diff
@@ -79,12 +68,35 @@ export class StartupPageContribution implements IWorkbenchContribution {
 				}
 			}
 		);
+	}
+}
+
+export class StartupPageRunnerContribution implements IWorkbenchContribution {
 
+	static readonly ID = 'workbench.contrib.startupPageRunner';
+
+	constructor(
+		@IConfigurationService private readonly configurationService: IConfigurationService,
+		@IEditorService private readonly editorService: IEditorService,
+		@IWorkingCopyBackupService private readonly workingCopyBackupService: IWorkingCopyBackupService,
+		@IFileService private readonly fileService: IFileService,
+		@IWorkspaceContextService private readonly contextService: IWorkspaceContextService,
+		@ILifecycleService private readonly lifecycleService: ILifecycleService,
+		@IWorkbenchLayoutService private readonly layoutService: IWorkbenchLayoutService,
+		@IProductService private readonly productService: IProductService,
+		@ICommandService private readonly commandService: ICommandService,
+		@IWorkbenchEnvironmentService private readonly environmentService: IWorkbenchEnvironmentService,
+		@IStorageService private readonly storageService: IStorageService,
+		@ILogService private readonly logService: ILogService,
+		@INotificationService private readonly notificationService: INotificationService
+	) {
 		this.run().then(undefined, onUnexpectedError);
 	}
 
 	private async run() {
 
+		await this.lifecycleService.when(LifecyclePhase.Restored);
```

Comment:
Maybe for the future, add a comment as to why that is needed.

Reply:
Was just pushing my comment commit :)